frontend/sw360-portlet/src/main/java/com/siemens/sw360/portal/portlets/components/ComponentPortlet.java

```diff
@@ -796,17 +804,25 @@ public void updateRelease(ActionRequest request, ActionResponse response) throws
                     release.setComponentId(component.getId());
                     release.setClearingState(ClearingState.NEW_CLEARING);
                     ComponentPortletUtils.updateReleaseFromRequest(request, release);
-                    releaseId = client.addRelease(release, user);
-
-                    if (releaseId != null) {
-                        response.setRenderParameter(RELEASE_ID, releaseId);
-                        String successMsg = "Release " + printName(release) + " added successfully";
-                        SessionMessages.add(request, "request_processed", successMsg);
-                    } else {
-                        setSW360SessionError(request, ErrorMessages.RELEASE_NOT_ADDED);
+                    AddDocumentRequestSummary summary = client.addRelease(release, user);
+
+                    AddDocumentRequestStatus status = summary.getRequestStatus();
+                    switch(status){
+                        case SUCCESS:
+                            response.setRenderParameter(RELEASE_ID, summary.getId());
+                            String successMsg = "Release " + printName(release) + " added successfully";
+                            SessionMessages.add(request, "request_processed", successMsg);
+                            response.setRenderParameter(PAGENAME, PAGENAME_EDIT_RELEASE);
+                            break;
+                        case DUPLICATE:
+                            setSW360SessionError(request, ErrorMessages.RELEASE_DUPLICATE);
+                            response.setRenderParameter(PAGENAME, PAGENAME_DETAIL);
```

Comment:
So if a duplicate exists for whatever reason (mistyped version, for example), the user gets thrown back to the component display and loses any data entered in the release edit form. Same effect with projects and components. In contrast, license edit form retains entered information when there's a duplicate and lets the user correct his input.